File: leetcode/226-翻转二叉树/226-翻转二叉树.py
# Definition for a binary tree node.
from re import S


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


# 226-翻转二叉树
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        queue = [root]
        while len(queue) > 0:
            size = len(queue)
            for i in range(size):
                node = queue[0]
                queue = queue[1:]
                node.left,node.right = node.right,node.left
                if not node.left is None:
                    queue.append(node.left)
                if not node.rigt is None:
                    queue.append(node.right)
        return root

File: leetcode/226-翻转二叉树/226-翻转二叉树.js
/**
 * Definition for a binary tree node.
*/
function TreeNode(val, left, right) {
    this.val = (val === undefined ? 0 : val)
    this.left = (left === undefined ? null : left)
    this.right = (right === undefined ? null : right)
}

/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */

// 226-翻转二叉树
var invertTree = function (root) {
    if (!root) {
        return null;
    }
    const dfs = function (node) {
        if (!node) {
            return;
        }
        const tmp = node.left;
        node.left = node.right;
        node.right = tmp;
        dfs(node.left);
        dfs(node.right);
    };
    dfs(root);
    return root;
};